tgui/packages/tgui/interfaces/common/AtmosHandbook.tsx

```diff
@@ -0,0 +1,242 @@
+import { Component, InfernoNode } from 'inferno';
+import { useBackend, useSharedState } from '../../backend';
+import {
+  Section,
+  LabeledList,
+  Tooltip,
+  Stack,
+  Flex,
+  Box,
+  Button,
+  Input,
+} from '../../components';
+
+/**
+ * This describes something that influences a particular reaction
+ * E.G.
+ * factor_id = 'oxygen'
+ * factor_id_type = 'gas'
+ * factor_name = 'Oxygen
+ * reaction_id = 'plasmafire'
+ * desc = 'Influences the burn rate and consumption ratio.'
+ */
+
+type Factor = {
+  factor_id?: string;
+  factor_type: string; // 'gas' or 'misc', for now.
+  factor_name: string;
+
+  desc: string;
+  tooltip?: string;
+};
+
+type Reaction = {
+  id: string;
+  name: string;
+  description: string;
+  factors: Factor[];
+};
+
+type Gas = {
+  id: string;
+  name: string;
+  description: string;
+  specific_heat: number;
+  reactions: { [key: string]: string } | [];
+};
+
+class ChangingInput extends Component<
+  {
+    title: InfernoNode | string;
+    icon: string;
+    onChange: (inputValue: string) => void;
+  },
+  { activeInput: boolean }
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeInput: false,
+    };
+  }
+
+  modifiedOnChange(value) {
+    this.setState({ activeInput: false });
+    this.props.onChange(value);
+  }
+
+  toggleInput() {
+    this.setState((prevState) => ({
+      activeInput: !prevState.activeInput,
+    }));
+  }
+
+  render() {
+    const { title, icon } = this.props;
+    return (
+      <Flex align="center">
+        <Flex.Item grow>
+          {this.state?.activeInput ? (
+            <Input
+              fluid
+              onChange={(e, value) => this.modifiedOnChange(value)}
+            />
+          ) : (
+            title
+          )}
+        </Flex.Item>
+        <Flex.Item>
+          <Button icon={icon} onClick={() => this.toggleInput()} />
+        </Flex.Item>
+      </Flex>
+    );
+  }
+}
+
+const GasHandbook = (props: { gasTextBook: Gas[] }, context) => {
+  const { act, data } = useBackend(context);
+  const { gasTextBook } = props;
+  const [activeGasId, setActiveGasId] = useSharedState(
+    context,
+    'activeGasId',
+    ''
+  );
+  const [activeReactionId, setActiveReactionId] = useSharedState(
+    context,
+    'activeReactionId',
+    ''
+  );
+  const relevantGas = gasTextBook.find((gas) => gas.id === activeGasId);
+  return (
+    <Section
+      title={
+        <ChangingInput
+          title={relevantGas ? 'Gas: ' + relevantGas.name : 'Gas Lookup'}
+          icon="search"
+          onChange={(keyword) =>
+            setActiveGasId(
+              gasTextBook.find((gas) =>
+                gas.name.toLowerCase().startsWith(keyword.toLocaleLowerCase())
```

Comment:
Why `toLocaleLowerCase` instead of `toLowerCase`? I10n shouldn't matter here.

Reply:
i have no idea what localelowercase it, must've slipped when it autosuggested.